runtime/src/main/java/org/corfudb/runtime/CheckpointLivenessUpdater.java

```diff
@@ -52,24 +58,31 @@ private Table<TableName, ActiveCPStreamMsg, Message> getActiveCheckpointsTable()
                 TableOptions.fromProtoSchema(ActiveCPStreamMsg.class));
     }
 
-    private void addTask() {
-        if (!currentTable.isPresent()) {
+    private void updateHeartbeat() {
+        TableName table;
+        try {
+            table = currentTable.get();
```

Comment:
currentTable should be made volatile to ensure that this thread always reads the most recent value.